### PR TITLE
Moving Sequence to Task :RecoLocalMuon RecoMuon RecoEgamma RecoEcal RecoHI

### PR DESCRIPTION
**Pull request by jeongeun**
Move Sequence to Task for RecoLocalMuon/RecoMuon/RecoEgamma/RecoEcal/RecoHI  config files
(An extension of the previous task [PR#27474](https://github.com/cms-sw/cmssw/pull/27474))

Tested in CMSSW_11_0_X_2019-07-28-2300 release, The PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html).

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
+upgrade

File: RecoEcal/EgammaClusterProducers/python/particleFlowSuperClusteringSequence_cff.py [hunk 1]
```diff
@@ -8,7 +8,8 @@
 # Producer for energy corrections
 #from RecoEcal.EgammaClusterProducers.correctedDynamicHybridSuperClusters_cfi import *
 # PFECAL super clusters, either hybrid-clustering clone (Box) or mustache.
-particleFlowSuperClusteringSequence = cms.Sequence(particleFlowSuperClusterECAL)
+particleFlowSuperClusteringTask = cms.Task(particleFlowSuperClusterECAL)
+#particleFlowSuperClusteringSequence = cms.Sequence(particleFlowSuperClusterECAL)
```
Comment (perrotta) [inline RecoEcal/EgammaClusterProducers/python/particleFlowSuperClusteringSequence_cff.py]: Either remove (preferred, as it is not needed) or uncomment (if you want to maintain a sequence with the same name as the file).
If uncommented, please use
```PYTHON
particleFlowSuperClusteringSequence = cms.Sequence(particleFlowSuperClusteringTask)
```

File: RecoMuon/Configuration/python/RecoMuonCosmics_cff.py [hunk 2]
```diff
@@ -45,12 +45,14 @@
 ## Sequences
 
 # Stand Alone Tracking
-STAmuontrackingforcosmics = cms.Sequence(CosmicMuonSeed*cosmicMuons)
+STAmuontrackingforcosmicsTask = cms.Task(CosmicMuonSeed,cosmicMuons)
+STAmuontrackingforcosmics = cms.Sequence(STAmuontrackingforcosmicsTask)
 # Stand Alone Tracking plus muon ID
-STAmuonrecoforcosmics = cms.Sequence(STAmuontrackingforcosmics)
+STAmuonrecoforcosmics = cms.Sequence(STAmuontrackingforcosmicsTask)
```
Comment (perrotta) [inline RecoMuon/Configuration/python/RecoMuonCosmics_cff.py]: This line is not needed any more

File: RecoMuon/Configuration/python/RecoMuonCosmics_cff.py [hunk 3]
```diff
@@ -72,11 +74,18 @@
...
+allmuons = cms.Sequence(allmuonsTask)
 
 # Final sequence
-muonrecoforcosmics = cms.Sequence(muontrackingforcosmics*allmuons*muonsFromCosmics)
-muonRecoAllGR = cms.Sequence(muonrecoforcosmics)
+muonrecoforcosmicsTask = cms.Task(muontrackingforcosmicsTask,
+                                  allmuonsTask,
+                                  muonsFromCosmics)
+muonrecoforcosmics = cms.Sequence(muonrecoforcosmicsTask)
+#muonRecoAllGR = cms.Sequence(muonrecoforcosmicsTask)
```
Comment (perrotta) [inline RecoMuon/Configuration/python/RecoMuonCosmics_cff.py]: This line can be removed

File: RecoMuon/Configuration/python/RecoMuonCosmics_cff.py [hunk 4]
```diff
@@ -152,22 +158,26 @@
...
+muontrackingforcosmicsWitht0CorrectionTask = cms.Task(STAmuontrackingforcosmicsWitht0CorrectionTask,globalCosmicMuonsWitht0Correction)
+muontrackingforcosmicsWitht0Correction = cms.Sequence(muontrackingforcosmicsWitht0CorrectionTask)
 
 # Stand Alone Tracking plus muon ID
-STAmuonrecoforcosmicsWitht0Correction = cms.Sequence(STAmuontrackingforcosmicsWitht0Correction)
+STAmuonrecoforcosmicsWitht0Correction = cms.Sequence(STAmuontrackingforcosmicsWitht0CorrectionTask)
 
 # all muons id
-allmuonsWitht0Correction = cms.Sequence(muonsWitht0Correction)
+#allmuonsWitht0Correction = cms.Sequence(muonsWitht0Correction)
```
Comment (perrotta) [inline RecoMuon/Configuration/python/RecoMuonCosmics_cff.py]: remove, as not needed

File: RecoMuon/Configuration/python/RecoMuonCosmics_cff.py [hunk 5]
```diff
@@ -232,16 +246,18 @@
 #Sequences
 
 # Stand Alone Tracking
-STAmuontrackingforcosmicsNoRPC = cms.Sequence(cosmicMuonsNoRPC)
+#STAmuontrackingforcosmicsNoRPC = cms.Sequence(cosmicMuonsNoRPC)
```
Comment (perrotta) [inline RecoMuon/Configuration/python/RecoMuonCosmics_cff.py]: remove

File: RecoMuon/Configuration/python/RecoMuonCosmics_cff.py [hunk 6]
```diff
@@ -232,16 +246,18 @@
...
 
 # Stand Alone Tracking plus global tracking
-muontrackingforcosmicsNoRPC = cms.Sequence(STAmuontrackingforcosmicsNoRPC*globalCosmicMuonsNoRPC)
+muontrackingforcosmicsNoRPCTask = cms.Task(cosmicMuonsNoRPC,globalCosmicMuonsNoRPC)
+muontrackingforcosmicsNoRPC = cms.Sequence(muontrackingforcosmicsNoRPCTask)
 
 # all muons id
-allmuonsNoRPC = cms.Sequence(muonsNoRPC)
+#allmuonsNoRPC = cms.Sequence(muonsNoRPC)
```
Comment (perrotta) [inline RecoMuon/Configuration/python/RecoMuonCosmics_cff.py]: remove

**Comment by fabiocos** [comment]
@jeongeun when do you expect to address the comments?

File: RecoMuon/Configuration/python/RecoMuonCosmics_cff.py [hunk 7]
```diff
@@ -81,11 +79,7 @@
 allmuons = cms.Sequence(allmuonsTask)
 
 # Final sequence
-muonrecoforcosmicsTask = cms.Task(muontrackingforcosmicsTask,
-                                  allmuonsTask,
-                                  muonsFromCosmics)
-muonrecoforcosmics = cms.Sequence(muonrecoforcosmicsTask)
-#muonRecoAllGR = cms.Sequence(muonrecoforcosmicsTask)
+muonrecoforcosmicsTask = cms.Task(muontrackingforcosmicsTask)
```
Comment (perrotta) [inline RecoMuon/Configuration/python/RecoMuonCosmics_cff.py]: Why did you remove allmuonsTask and muonsFromCosmics ?

File: RecoMuon/Configuration/python/RecoMuonCosmics_cff.py [hunk 8]
```diff
@@ -102,19 +105,16 @@
...
 # Stand Alone Tracking plus global tracking
-muontrackingforcosmics1Leg = cms.Sequence(STAmuontrackingforcosmics1Leg*globalCosmicMuons1Leg)
-
-# all muons id
-allmuons1Leg = cms.Sequence(muons1Leg)
+muontrackingforcosmics1LegTask = cms.Task(STAmuontrackingforcosmics1LegTask, globalCosmicMuons1Leg)
 
 # Stand Alone Tracking plus muon ID
-STAmuonrecoforcosmics1Leg = cms.Sequence(STAmuontrackingforcosmics1Leg)
+STAmuonrecoforcosmics1Leg = cms.Sequence(STAmuontrackingforcosmics1LegTask)
```
Comment (perrotta) [inline RecoMuon/Configuration/python/RecoMuonCosmics_cff.py]: This (and the comment above) can probably be also cleaned out: it is identical to STAmuontrackingforcosmics1Leg (i.e. there is not the assumed muon id) and it is never used anywhere

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Migration from Sequences to Tasks implemented for the packages RecoLocalMuon, RecoMuon, RecoEgamma, RecoEcal, RecoHI
- Jenkins tests passs and show no differences

**Comment by fabiocos** [comment]
@kpedro88 you had already signed this PR before its update requested by @perrotta , it looks ok to me, I will integrate it

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Comment by kpedro88** [comment]
+upgrade
